lib/Driver.js

```diff
@@ -83,80 +84,81 @@ module.exports = class Driver {
    * @returns {Promise<DidDocument>}
    */
   async generate({keyType = 'Ed25519VerificationKey2018'} = {}) {
-    const publicKey = await LDKeyPair.generate({type: keyType});
-    return this.keyToDidDoc(publicKey);
+    const keyPair = await cryptoLd.generate({type: keyType});
+    return this.keyPairToDidDocument({keyPair});
   }
 
   /**
    * Converts an Ed25519KeyPair object to a `did:key` method DID Document.
    *
-   * @param {Ed25519KeyPair} edKey
-   * @returns {DidDocument}
+   * @param {Ed25519KeyPair} keyPair - The key pair to use to generate the
+   *   DID Document.
+   * @returns {object} - An object containing the `didDocument` and `keyPairs`.
    */
-  async keyToDidDoc(edKey) {
-    const did = `did:key:${edKey.fingerprint()}`;
-    const keyId = `${did}#${edKey.fingerprint()}`;
-    edKey.controller = did;
+  async keyPairToDidDocument({keyPair} = {}) {
+    const did = `did:key:${keyPair.fingerprint()}`;
+    const edKeyPair = await cryptoLd.from(keyPair);
+    edKeyPair.id = `${did}#${edKeyPair.fingerprint()}`;
+    edKeyPair.controller = did;
+
+    const dhKeyPair = await X25519KeyPair.fromEdKeyPair(edKeyPair);
+    dhKeyPair.id = `${did}#${dhKeyPair.fingerprint()}`;
+    dhKeyPair.controller = did;
 
-    const dhKey = await X25519KeyPair.fromEdKeyPair(edKey);
-    dhKey.id = `${did}#${dhKey.fingerprint()}`;
+    // get the public components of each keypair
+    const publicEdKey = await edKeyPair.export({publicKey: true});
+    const publicDhKey = await dhKeyPair.export({publicKey: true});
 
-    const didDoc = {
+    // FIXME: remove private key accidentally exported to public key
+    delete publicDhKey.privateKeyBase58;
```

Comment:
@dlongley @dmitrizagidulin -- `dhKeyPair.export({publicKey: true});` includes the private key, seems like a nasty bug, I tried to track it down but it wasn't obvious where it was happening due to all the class/subclass jumping. Anything obvious jumping out to either of your wrt. X25519KeyPair implementation?

Reply:
@msporny - I can't replicate this behavior. Just to be sure, I added explicit smoke tests for it, over at https://github.com/digitalbazaar/x25519-key-agreement-key-2019/pull/15, and they seem to pass with expected results.